vulture/config.py

```diff
@@ -0,0 +1,193 @@
+"""
+This module handles retrieval of configuration values from either the
+command-line arguments or the pyproject.toml file.
+"""
+from __future__ import print_function
+import argparse
+import sys
+from os.path import abspath, exists
+
+import toml
+
+from .version import __version__
+
+#: Possible configuration options and their respective defaults
+DEFAULTS = {
+    "min_confidence": 0,
+    "paths": [],
+    "exclude": [],
+    "ignore_decorators": [],
+    "ignore_names": [],
+    "make_whitelist": False,
+    "sort_by_size": False,
+    "verbose": False,
+}
+
+#: sentinel value to distinguish between "False" and "no default given"
+NO_DEFAULT = object()
+
+
+def from_dict(data):
+    """
+    Create a new config dictionary from an existing one, assign possible
+    defaults and warn about unprocessed options.
+    """
+    # keep a copy of the keys, so we can keep track of any unprocessed
+    # values.
+    remaining_keys = set(data.keys())
+
+    output = {}
+    for key, default in DEFAULTS.items():
+        output[key] = data.get(key, default)
+        remaining_keys.discard(key)
+    for remainder in sorted(remaining_keys):
+        print(
+            "Unprocessed config option {}".format(remainder), file=sys.stderr
+        )
```

Comment:
Done in 3a5802d

I feel a bit uncomfortable that I had to use `mock.patch` in the tests for this, but using DI here would make the code a bit more complicated (not overly so though), which is why I leaned towards monkeypatching. Let me know what you think about this.

Another point where I may have gone overboard is the unit-test for `sys.exit`. As it stands, the *test* is fairly tightly coupled to the implementation details of `sys.exit`. But I doubt that such a low-level and important method will ever change in Python. Considering how many OS-level system utilities are written in Python.

The reason the assertion of `sys.exit` call looks a bit complicated is because I aimed for a future-proof test. No matter how `sys.exit` will be called in the future, this assertion ensures that we never exit with a zero exit-code. Thoughts?